### PR TITLE
doc team edit

### PR DESCRIPTION
**Pull request by kersten1**
Very minor suggestions. LGTM!

### CODE REVIEW

File: src/riscv-double-trap.adoc [hunk 1]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
 The Smdbltrp extension introduces a read-write critical-error-trigger (`cetrig`)
 field (bit 19) to the `dcsr` CSR cite:[DBG]. When `cetrig` is set to 1, a hart
 in a critical error state enters Debug Mode instead of asserting the
-critical-error signal to the platform. Upon such entry into Debug Mode, the
+critical-error signal to the platform. When the hart enters Debug Mode, the
 `cause` field of `dcsr` is set to 7, and its `extcause` field is set to 0,
 indicating a critical error triggered the Debug Mode entry. This cause has the
-highest priority among all reasons for entering Debug Mode. Resuming from Debug
-Mode following an entry from the critical error state returns the hart to the
+highest priority among all reasons for entering Debug Mode. Resuming operations from
```
Comment (kersten1) [inline src/riscv-double-trap.adoc:231]: Not sure if "operations" is the right word here. The sentence would benefit from having a noun though.

File: src/riscv-double-trap.adoc [hunk 2]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
 [NOTE]
 ====
-The implication of this specification is that resuming from Debug Mode,
-following an entry due to a critical error, will either result in an immediate
+The implication of this specification is that resuming operations from Debug Mode,
+following an entry due to a critical error will either result in an immediate
 re-entry into Debug Mode due to the critical error if `cetrig` is set to 1, or
-it will cause the critical-error signal to be asserted to the platform.  The
-debugger may resume with `cetrig` set to 0 to allow the platform defined actions
+it will cause the critical-error signal to be sent to the platform.  The
```
Comment (kersten1) [inline src/riscv-double-trap.adoc:240]:
```suggestion
it will cause the critical-error signal to be asserted to the platform.  The
```

File: src/riscv-double-trap.adoc (hunk 1 above)
Comment (ved-rivos) [reply src/riscv-double-trap.adoc:231]: In this context "Resuming" is well understood.

File: src/riscv-double-trap.adoc [hunk 3]
```diff
@@ -137,12 +137,12 @@ An `SRET` instruction sets the `SDT` bit to 0.
 
 [NOTE]
 ====
-A trap handler after saving the state needed for resuming from the trap,
-including `scause`, `sepc`, and `stval` among others, should clear the `SDT` bit
-when it is reentrant.
+After a trap handler, such as `scause`, `sepc`, and `stval` 
+among others, saves the state that is needed to resume from the trap, it
+should clear the `SDT` bit when it is reentrant.
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:142]: This change  is not right. The `scause` `sepc` are the   state  that the guideline is stating must be saved before clearing the `SDT` bit. Dont see a need to change this sentence.

File: src/riscv-double-trap.adoc [hunk 4]
```diff
@@ -137,12 +137,12 @@ An `SRET` instruction sets the `SDT` bit to 0.
...
 
-Resetting of the `SDT` by an `SRET` enables the trap handler to detect double
-trap occuring during the tail phase, where it restores critical state to return
+If the `SDT` is reset by an `SRET`, the trap handler is enabled to detect double
+traps that occur during the tail phase, where it restores critical state to return
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:145]: This is explaining why the ISA decided to reset the `SDT`. I think original sentence is good.

File: src/riscv-double-trap.adoc [hunk 5]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
+the affected hart or restarting the entire platform, among others.
 ====
 
 An `MRET` instruction sets the `MDT` bit to 0.
 
 The Smdbltrp extension introduces a read-write critical-error-trigger (`cetrig`)
 field (bit 19) to the `dcsr` CSR cite:[DBG]. When `cetrig` is set to 1, a hart
 in a critical error state enters Debug Mode instead of asserting the
-critical-error signal to the platform. Upon such entry into Debug Mode, the
+critical-error signal to the platform. When the hart enters Debug Mode, the
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:228]: "Upon such"  is important since the following actions like setting cause to 7 do not occur on other debug mode entries. We should not change this.

File: src/riscv-double-trap.adoc [hunk 6]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
-Mode following an entry from the critical error state returns the hart to the
+highest priority among all reasons for entering Debug Mode. Resuming operations from 
+Debug Mode following an entry from the critical error state returns the hart to the
 critical error state.
 
 [NOTE]
 ====
-The implication of this specification is that resuming from Debug Mode,
-following an entry due to a critical error, will either result in an immediate
+The implication of this specification is that resuming operations from Debug Mode,
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:237]: We can keep it as is - resuming  from Debug  Mode - is a well defined thing.

File: src/riscv-double-trap.adoc [hunk 7]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
+following an entry due to a critical error will either result in an immediate
 re-entry into Debug Mode due to the critical error if `cetrig` is set to 1, or
 it will cause the critical-error signal to be asserted to the platform.  The
-debugger may resume with `cetrig` set to 0 to allow the platform defined actions
+debugger can resume with `cetrig` set to 0 to allow the platform-defined actions
 on critical-error signal to occur. Other possible actions include initiating a
-hart or platform reset using the Debug Module reset control.
+hart or resetting the platform by using the Debug Module reset control.
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:243]: The intent was to say "hart or platform" reset. With this  rewrite "initiating a hart" has no meaning.

Comment (kersten1) [inline src/riscv-double-trap.adoc:243]:
```suggestion
hart or platform by using the Debug Module reset control.
```

File: src/riscv-double-trap.adoc (hunk 4 above)
Comment (kersten1) [reply src/riscv-double-trap.adoc:145]: Starting a sentence with a gerund can be confusing for translation. This sentence is also passive so making it more active can help with understanding. Can we compromise with this? "Resetting the `SDT` by an `SRET` enables the trap handler to detect a double trap that occurs during the tail phase, where it restores a critical state to return..."?

File: src/riscv-double-trap.adoc (hunk 5 above)
Comment (kersten1) [inline src/riscv-double-trap.adoc:228]:
```suggestion
critical-error signal to the platform. Upon such entry into  Debug Mode, the
```

File: src/riscv-double-trap.adoc (hunk 1 above)
Comment (kersten1) [reply src/riscv-double-trap.adoc:231]: Yes but it would benefit from having a noun. In the original, it is not clear if "resuming" is a noun or a verb. Adding a noun would make it a verb, which is then easier to translate. I will cave on this one if you really want the original though.

File: src/riscv-double-trap.adoc (hunk 6 above)
Comment (kersten1) [reply src/riscv-double-trap.adoc:237]: Same with this one. The noun makes it clearer.

File: src/riscv-double-trap.adoc (hunk 3 above)
Comment (kersten1) [reply src/riscv-double-trap.adoc:142]: The original in this case is hard to parse though. Can we compromise with "After a trap handler saves the state, such as `scause`, `sepc`, and `stval`, that is needed to resume from the trap, it should clear the `SDT` bit...."

File: src/riscv-double-trap.adoc (hunk 1 above)
Comment (ved-rivos) [reply src/riscv-double-trap.adoc:231]: Yes. We should retain the original.

File: src/riscv-double-trap.adoc [hunk 8]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
 The Smdbltrp extension introduces a read-write critical-error-trigger (`cetrig`)
 field (bit 19) to the `dcsr` CSR cite:[DBG]. When `cetrig` is set to 1, a hart
 in a critical error state enters Debug Mode instead of asserting the
-critical-error signal to the platform. Upon such entry into Debug Mode, the
+critical-error signal to the platform. Upon such entry into  Debug Mode, the
 `cause` field of `dcsr` is set to 7, and its `extcause` field is set to 0,
 indicating a critical error triggered the Debug Mode entry. This cause has the
-highest priority among all reasons for entering Debug Mode. Resuming from Debug
-Mode following an entry from the critical error state returns the hart to the
+highest priority among all reasons for entering Debug Mode. Resuming operations from
```
Comment (kersten1) [inline src/riscv-double-trap.adoc:231]:
```suggestion
highest priority among all reasons for entering Debug Mode. Resuming from Debug
```

File: src/riscv-double-trap.adoc [hunk 9]
```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
...
 The Smdbltrp extension introduces a read-write critical-error-trigger (`cetrig`)
 field (bit 19) to the `dcsr` CSR cite:[DBG]. When `cetrig` is set to 1, a hart
 in a critical error state enters Debug Mode instead of asserting the
-critical-error signal to the platform. Upon such entry into Debug Mode, the
+critical-error signal to the platform. Upon such entry into  Debug Mode, the
 `cause` field of `dcsr` is set to 7, and its `extcause` field is set to 0,
 indicating a critical error triggered the Debug Mode entry. This cause has the
 highest priority among all reasons for entering Debug Mode. Resuming from Debug
-Mode following an entry from the critical error state returns the hart to the
+Debug Mode following an entry from the critical error state returns the hart to the
```
Comment (kersten1) [inline src/riscv-double-trap.adoc:232]:
```suggestionMode following an entry from the critical error state returns the hart to the
```

Comment (kersten1) [inline src/riscv-double-trap.adoc:232]:
```suggestion
Mode following an entry from the critical error state returns the hart to the
```

File: src/riscv-double-trap.adoc [hunk 10]
```diff
@@ -200,32 +200,32 @@ _unexpected trap_ is as follows:
...
+the affected hart or restarting the entire platform, among others.
 ====
 
 An `MRET` instruction sets the `MDT` bit to 0.
 
 The Smdbltrp extension introduces a read-write critical-error-trigger (`cetrig`)
 field (bit 19) to the `dcsr` CSR cite:[DBG]. When `cetrig` is set to 1, a hart
 in a critical error state enters Debug Mode instead of asserting the
-critical-error signal to the platform. Upon such entry into Debug Mode, the
+critical-error signal to the platform. Upon such entry into  Debug Mode, the
```
Comment (kersten1) [inline src/riscv-double-trap.adoc:228]:
```suggestion
critical-error signal to the platform. Upon such entry into Debug Mode, the
```

File: src/riscv-double-trap.adoc [hunk 11]
```diff
@@ -234,13 +234,13 @@ critical error state.
 
 [NOTE]
 ====
-The implication of this specification is that resuming from Debug Mode,
-following an entry due to a critical error, will either result in an immediate
+The implication of this specification is that resuming operations from Debug Mode,
```
Comment (kersten1) [inline src/riscv-double-trap.adoc:237]:
```suggestion
The implication of this specification is that resuming from Debug Mode,
```

File: src/riscv-double-trap.adoc (hunk 4 above)
Comment (ved-rivos) [reply src/riscv-double-trap.adoc:145]: Yes. That would be good.

File: src/riscv-double-trap.adoc (hunk 3 above)
Comment (ved-rivos) [reply src/riscv-double-trap.adoc:142]: We could update to:
"After a trap handler has saved the state, such as `scause`, `sepc`, and `stval`, needed for resuming from the trap and is reentrant it should clear the `SDT` bit."

File: src/riscv-double-trap.adoc [hunk 12]
```diff
@@ -137,13 +137,13 @@ An `SRET` instruction sets the `SDT` bit to 0.
...
-including `scause`, `sepc`, and `stval` among others, should clear the `SDT` bit
-when it is reentrant.
+After a trap handler has saved the state, such as `scause`, `sepc`, 
+and `stval`, needed for resuming from the trap and is reentrant, it 
+should clear the `SDT` bit.
 
-Resetting of the `SDT` by an `SRET` enables the trap handler to detect double
-trap occuring during the tail phase, where it restores critical state to return
-from a trap.
+Resetting the SDT by an SRET enables the trap handler to detect a double
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:144]: @kersten1 the backticks around SDT are missing. Also could we change "that occurs" to "that may occur". The double trap may or may not occur - most likely does not.

File: src/riscv-double-trap.adoc [hunk 13]
```diff
@@ -137,13 +137,13 @@ An `SRET` instruction sets the `SDT` bit to 0.
...
-when it is reentrant.
+After a trap handler has saved the state, such as `scause`, `sepc`, 
+and `stval`, needed for resuming from the trap and is reentrant, it 
+should clear the `SDT` bit.
 
-Resetting of the `SDT` by an `SRET` enables the trap handler to detect double
-trap occuring during the tail phase, where it restores critical state to return
-from a trap.
+Resetting the SDT by an SRET enables the trap handler to detect a double 
+trap that occurs during the tail phase, where it restores a critical state
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:145]: @kersten1 "a critical state" -> "critical state" since there are many such critical states that are restored.

File: src/riscv-double-trap.adoc (hunk 12 above)
Comment (kersten1) [inline src/riscv-double-trap.adoc:144]:
```suggestion
Resetting the `SDT` by an SRET enables the trap handler to detect a double 
```

File: src/riscv-double-trap.adoc [hunk 14]
```diff
@@ -235,12 +235,12 @@ critical error state.
...
 The implication of this specification is that resuming from Debug Mode,
-following an entry due to a critical error, will either result in an immediate
+following an entry due to a critical error will either result in an immediate
 re-entry into Debug Mode due to the critical error if `cetrig` is set to 1, or
 it will cause the critical-error signal to be asserted to the platform.  The
-debugger may resume with `cetrig` set to 0 to allow the platform defined actions
+debugger can resume with `cetrig` set to 0 to allow the platform-defined actions
 on critical-error signal to occur. Other possible actions include initiating a
-hart or platform reset using the Debug Module reset control.
+hart or platform by using the Debug Module reset control.
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:243]: @kersten1 "hart of platform by using" should be "hart of platform reset using"

Comment (kersten1) [inline src/riscv-double-trap.adoc:243]:
```suggestion
hart or platform using the Debug Module reset control.
```

File: src/riscv-double-trap.adoc [hunk 15]
```diff
@@ -240,7 +240,7 @@ re-entry into Debug Mode due to the critical error if `cetrig` is set to 1, or
 it will cause the critical-error signal to be asserted to the platform.  The
 debugger can resume with `cetrig` set to 0 to allow the platform-defined actions
 on critical-error signal to occur. Other possible actions include initiating a
-hart or platform by using the Debug Module reset control.
+hart or platform using the Debug Module reset control.
```
Comment (ved-rivos) [inline src/riscv-double-trap.adoc:243]: @kersten1 
The sentence is not reading right. It says:
"Other possible actions include initiating a hart or platform using the Debug Module reset control."
It should say:
"Other possible actions include initiating a hart or platform reset using the Debug Module reset control."

Comment (ved-rivos) [reply src/riscv-double-trap.adoc:243]: @kersten1 - I will merge this and fix this post merge.
